datadog_lambda/tracing.py

```diff
@@ -89,9 +91,33 @@ def _context_obj_to_headers(obj):
     }
 
 
-def extract_dd_trace_context(event):
+def extract_dd_trace_context_from_context(context):
+    if context.client_context is not None:
+        client_context_json = base64.b64decode(context.client_context).decode("utf-8")
+        client_context_object = json.loads(client_context_json)
+        trace_data = client_context_object.get("custom", {}).get("_datadog", {})
+        ctx_trace_id = trace_data.get(TraceHeader.TRACE_ID)
+        ctx_parent_id = trace_data.get(TraceHeader.PARENT_ID)
+        ctx_sampling_priority = trace_data.get(TraceHeader.SAMPLING_PRIORITY)
+
+        return ctx_trace_id, ctx_parent_id, ctx_sampling_priority
+    return None, None, None
+
+
+def extract_dd_trace_context_from_event(event):
+    headers = event.get("headers", {})
+    lowercase_headers = {k.lower(): v for k, v in headers.items()}
+    trace_id = lowercase_headers.get(TraceHeader.TRACE_ID)
+    parent_id = lowercase_headers.get(TraceHeader.PARENT_ID)
+    sampling_priority = lowercase_headers.get(TraceHeader.SAMPLING_PRIORITY)
+
+    return trace_id, parent_id, sampling_priority
+
+
+def extract_dd_trace_context(event, context):
     """
-    Extract Datadog trace context from the Lambda `event` object.
+    Extract Datadog trace context from the Lambda `event` object or
```

Comment:
Nit: I would just say extract from the `event` or `context` objects. Technically `client_context` is a key within the context object.

Reply:
Done!